scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -172,6 +172,15 @@ object TypedPipe extends Serializable {
 
    case class CoGroupedPipe[K, V](cogrouped: CoGrouped[K, V]) extends TypedPipe[(K, V)]
    case class ReduceStepPipe[K, V1, V2](reduce: ReduceStep[K, V1, V2]) extends TypedPipe[(K, V2)]
+
+
+   implicit class InvariantTypedPipe[T](val pipe: TypedPipe[T]) extends AnyVal {
+      /**
+       * If any errors happen below this line, but before a groupBy, write to a TypedSink
+       */
+      def addTrap(trapSink: Source with TypedSink[T])(implicit conv: TupleConverter[T]): TypedPipe[T] =
```

Comment:
this is a very rarely used method but it was giving me trouble with the two type parameters. I think this is a cleaner implementation anyway since it makes it clear that it returns the exact same type as the original pipe.

Reply:
+1